datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -270,37 +270,27 @@ fn not_in_list_utf8<OffsetSize: OffsetSizeTrait>(
 }
 
 //check all filter values of In clause are static.
-//include `CastExpr + Literal` or `Literal`
-fn check_all_static_filter_expr(list: &[Arc<dyn PhysicalExpr>]) -> bool {
-    list.iter().all(|v| {
-        let cast = v.as_any().downcast_ref::<expressions::CastExpr>();
-        if let Some(c) = cast {
-            c.expr()
-                .as_any()
-                .downcast_ref::<expressions::Literal>()
-                .is_some()
-        } else {
-            let cast = v.as_any().downcast_ref::<expressions::Literal>();
-            cast.is_some()
-        }
+//include `TryCastExpr/CastExpr + Literal` or `Literal`
+fn check_all_static_filter_expr(list: &[Arc<dyn PhysicalExpr>], schema: &Schema) -> bool {
+    let batch = RecordBatch::new_empty(Arc::new(schema.to_owned()));
```

Comment:
This is looking better -- but now `evaluate()` will be called *twice* on the inputs; Both here as well as in `cast_static_filter_to_set` . Since calling `evaluate` may be quite expensive, what do you think about calling it once and then passing the resulting `Vec<ColumnarValue>` to both `check_all_static_filter_expr` and `cast_static_filter_to_set`?

Reply:
I add a new function `try_cast_static_filter_to_set` to replace these two function in the InLIst.